contracts/v2/sovereignChains/GlobalExitRootManagerL2SovereignChain.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity 0.8.20;
+import "../../PolygonZkEVMGlobalExitRootL2.sol";
+
+/**
+ * Contract responsible for managing the exit roots for the Sovereign chains and global exit roots
+ */
+contract GlobalExitRootManagerL2SovereignChain is PolygonZkEVMGlobalExitRootL2 {
+    /**
+     * @dev Emitted when a new global exit root is inserted
+     */
+    event InsertGlobalExitRoot(bytes32 indexed newGlobalExitRoot);
```

Comment:
I would also add that this functionality will allow to prune the state, so you can better control the increasing

Reply:
@invocamanman thoughts on this ?
I'm thinking this feature as a mechanism in `v0.2.0` aggLayer version to avoid: set wrong GER, users claims from that LER continuously and `pp` cannot be generated.
@ignasirv genesis tool should include a flag (like `sovereignChain: true`) in order to create a genesis with those contracts